modules/currency-l0/src/main/scala/org/tessellation/currency/l0/snapshot/services/SentStateChannelBinaryTrackingService.scala

```diff
@@ -0,0 +1,76 @@
+package org.tessellation.currency.l0.snapshot.services
+
+import cats.Show
+import cats.effect.{Async, Ref}
+import cats.syntax.all._
+
+import scala.collection.immutable.ListMap
+
+import org.tessellation.currency.l0.node.IdentifierStorage
+import org.tessellation.schema.GlobalIncrementalSnapshot
+import org.tessellation.security.signature.Signed
+import org.tessellation.statechannel.StateChannelSnapshotBinary
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.{NonNegInt, PosInt}
+
+case class TrackingCheckResult(
+  confirmed: Set[Signed[StateChannelSnapshotBinary]],
+  pending: Map[Signed[StateChannelSnapshotBinary], NonNegInt],
+  toRetry: Set[Signed[StateChannelSnapshotBinary]]
+)
+
+object TrackingCheckResult {
+  implicit val show: Show[TrackingCheckResult] = (t: TrackingCheckResult) =>
+    s"TrackingCheckResult(confirmed=${t.confirmed}, pending=${t.pending}, toRetry=${t.toRetry})"
+}
+
+trait SentStateChannelBinaryTrackingService[F[_]] {
+  def setPending(binary: Signed[StateChannelSnapshotBinary]): F[Unit]
+  def getPending: F[ListMap[Signed[StateChannelSnapshotBinary], NonNegInt]]
+  def getRetriable: F[Option[Signed[StateChannelSnapshotBinary]]]
+  def updateByGlobalSnapshot(globalSnapshot: GlobalIncrementalSnapshot): F[Unit]
+}
+
+object SentStateChannelBinaryTrackingService {
+  private val maxChecks: PosInt = 3
```

Comment:
I am wondering if we should move it to config and what should be the correct number. It basically says for how many ordinals it waits until it retries.

Reply:
Shouldn't it be called something like 'retryOrdinalDelay' or similar?